signal.c

```diff
@@ -107,8 +113,48 @@ static const struct eventop evsigops = {
 /* Lock for evsig_base and evsig_base_n_signals_added fields. */
 static void *evsig_base_lock = NULL;
 #endif
-/* The event base that's currently getting informed about signals. */
+
+#if defined(_MSC_VER)
+// Microsoft explicitly excludes stdatomics.h from the list of supported
+// C11 features on Windows.
+// it also doesn't have INTPTR_WIDTH or __INTPTR_WIDTH__ with the preprocessor.
+// to be be fair, Linux also doesn't have INTPTR_WIDTH that early.
+// comparing sizeof(void *) also doesn't work with #if
+
+static inline void *atomic_load(intptr_t volatile *p)
+{
+	// atomic_load uses memory_order_seq_cst, so synchronize here as well
+	intptr_t x;
+	intptr_t comparand = *p;
+	do {
+		x = comparand;
+		if (sizeof(*p) == 8)
```

Comment:
Better to detect this at configuration time (cmake/autotools):
- for cmake there is already https://github.com/libevent/libevent/blob/9e346936d47a3a461f8ebaedabb2975bef3babf5/CMakeLists.txt#L709
- but it is not enough since it uses `#cmakedefine`, while it should use `@EVENT__HAVE_UINTPTR_T@` here - https://github.com/libevent/libevent/blob/9e346936d47a3a461f8ebaedabb2975bef3babf5/event-config.h.cmake#L444
- and for autotools - https://github.com/libevent/libevent/blob/9e346936d47a3a461f8ebaedabb2975bef3babf5/configure.ac#L551 seems that `AC_CHECK_SIZEOF` should be used for `uintptr_t`